kolena/io.py

```diff
@@ -52,7 +52,9 @@ def _deserialize_dataobject(x: Any) -> Any:
 
 
 def _serialize_dataobject_str(x: Any) -> Any:
-    return json.dumps(x, cls=DataObjectJSONEncoder)
+    if isinstance(x, list) or isinstance(x, dict):
```

Comment:
Can we add tests to prevent future regression?

Reply:
Yes. I am working on it

Reply:
Added a unit test for this method. After speaking with @yifanwu-kolena I'll be addressing the issue with `_deserialize_dataobject_str` calling `json.loads` for all strings when it should be more selective in a separate PR to limit the scope of this fix